boa/src/object/internal_methods.rs

```diff
@@ -390,6 +390,77 @@ impl GcObject {
         }
     }
 
+    /// Gets own property of 'Object'
+    ///
+    #[inline]
+    pub fn get_own_property(&self, key: &PropertyKey) -> Option<PropertyDescriptor> {
+        let _timer = BoaProfiler::global().start_event("Object::get_own_property", "object");
+
+        let object = self.borrow();
+        match object.data {
+            ObjectData::String(_) => self.string_exotic_get_own_property(key),
+            _ => self.ordinary_get_own_property(key),
+        }
+    }
+
+    /// StringGetOwnProperty abstract operation
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-stringgetownproperty
+    #[inline]
+    pub fn string_get_own_property(&self, key: &PropertyKey) -> Option<PropertyDescriptor> {
+        let object = self.borrow();
+
+        match key {
+            PropertyKey::Index(index) => {
+                let string = object.as_string().unwrap();
+                let pos = *index as usize;
+
+                if pos >= string.len() {
+                    return None;
+                }
+
+                let result_str = if let Some(utf16_val) = string.encode_utf16().nth(pos) {
+                    let chr_option = char::from_u32(u32::from(utf16_val));
+
+                    match chr_option {
+                        Some(chr) => Value::from(chr),
+                        None => Value::from(format!("\\u{:x}", utf16_val)),
+                    }
```

Comment:
I would simplify this a bit by using the `map_or_else` method in the `Option` type:
```suggestion
                    char::from_u32(u32::from(utf16_val)).map_or_else(|| Value::from(format!("\\u{:x}", utf16_val)), Value::from)
```

It probably requires a `cargo fmt`.

Reply:
Since if the `if let` on line 425 fails you just return `None` I think you can just use the `?` operator and just make the whole `if` one big `map`.